src/model/exercise.coffee

```diff
@@ -20,21 +26,22 @@ Exercises =
       url += "&entry.1091629000=#{BOOK_UID_XREF[options.bookUUID]}"
     url
 
-
-  getStimulus: (exercise) ->
-    exercise.content?.stimulus_html or ''
-
   getParts: (exercise) ->
     exercise.content?.questions or []
 
   isMultipart: (exercise) ->
     @getParts(exercise).length > 1
 
   hasInteractive: (exercise) ->
-    !!@getStimulus(exercise).match(/iframe.*(cnx.org|phet.colorado.edu)/)
+    exercise.has_interactive
 
   hasVideo: (exercise) ->
-    !!@getStimulus(exercise).match(/(youtube.com\/(v|embed)\/|khanacademy.org)/)
+    exercise.has_video
+
+  replacePlaceholders: (content) ->
```

Comment:
the `has_video` and `has_interactive` is still a big win.  at least we'll always get that right for the badges at the top and won't render a honking big sim by mistake